### PR TITLE
add: service version to caps for browserstack

### PR DESCRIPTION
**Pull request by Ankit098**
## Proposed changes
Adding a capability 'wdioService' in caps being sent to browserstack, to track the service being used.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [ ] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### Reviewers: @webdriverio/project-committers

<a href="https://gitpod.io/#https://github.com/webdriverio/webdriverio/pull/8350"></a>

### CODE REVIEW

File: packages/wdio-browserstack-service/src/launcher.ts [hunk 1]
```diff
@@ -8,6 +8,8 @@ import type { Capabilities, Services, Options } from '@wdio/types'
 import { BrowserstackConfig } from './types'
 
 const log = logger('@wdio/browserstack-service')
+// @ts-ignore
+import { bstackServiceVersion } from '../package.json'
```
Comment (christian-bromann) [inline packages/wdio-browserstack-service/src/launcher.ts:12]: What is `bstackServiceVersion` suppose to be? Are you referring to `version`, maybe this is what you wanted:
```suggestion
// @ts-ignore
import { version as bstackServiceVersion } from '../package.json'
```

**Review by christian-bromann — COMMENTED** [review]
Could we add a test for this as well?

File: packages/wdio-browserstack-service/src/launcher.ts (hunk 1 above)
Comment (Ankit098) [reply packages/wdio-browserstack-service/src/launcher.ts:12]: yes, will fix this and add test.

File: packages/wdio-browserstack-service/tests/launcher.test.ts [hunk 2]
```diff
@@ -9,6 +9,8 @@ const expect = global.expect as unknown as jest.Expect
 const log = logger('test')
 const error = new Error('I\'m an error!')
 const sleep = (ms: number = 100) => new Promise((resolve) => setTimeout(resolve, ms))
+// @ts-ignore
```
Comment (christian-bromann) [inline packages/wdio-browserstack-service/tests/launcher.test.ts:12]: This should not be needed.
```suggestion
```

File: packages/wdio-browserstack-service/src/launcher.ts [hunk 3]
```diff
@@ -8,6 +8,8 @@ import type { Capabilities, Services, Options } from '@wdio/types'
 import { BrowserstackConfig } from './types'
 
 const log = logger('@wdio/browserstack-service')
+// @ts-ignore
```
Comment (christian-bromann) [inline packages/wdio-browserstack-service/src/launcher.ts:11]:
```suggestion
```

File: packages/wdio-types/src/Capabilities.ts [hunk 4]
```diff
@@ -1082,6 +1082,7 @@ export interface BrowserStackCapabilities {
     browserName?: string
     browserVersion?: string
     acceptSslCerts?: boolean
+    wdioService?: string
```
Comment (christian-bromann) [inline packages/wdio-types/src/Capabilities.ts:1085]: This probably should be marked as `@private`
```suggestion
    wdioService?: string
```

**Review by christian-bromann — COMMENTED** [review]
Can we move the imports up where other imports are being made?

File: packages/wdio-browserstack-service/tests/launcher.test.ts (hunk 2 above)
Comment (Ankit098) [reply packages/wdio-browserstack-service/tests/launcher.test.ts:12]: Had added this as importing from `package.json` throws the `'rootDir' is expected to contain all source files.` error.

File: packages/wdio-browserstack-service/src/launcher.ts [hunk 5]
```diff
@@ -5,6 +5,8 @@ import * as BrowserstackLocalLauncher from 'browserstack-local'
 import logger from '@wdio/logger'
 import type { Capabilities, Services, Options } from '@wdio/types'
 
+// @ts-ignore
```
Comment (christian-bromann) [inline packages/wdio-browserstack-service/src/launcher.ts:8]: Why is this needed?

Comment (Ankit098) [reply packages/wdio-browserstack-service/src/launcher.ts:8]: Was seeing this error: `'rootDir' is expected to contain all source files. error` https://github.com/webdriverio/webdriverio/pull/8350#discussion_r875477783

Comment (christian-bromann) [reply packages/wdio-browserstack-service/src/launcher.ts:8]: Gotcha, yeah we can ignore this for now

**Review by christian-bromann — APPROVED** [review]
LGTM 👍
